espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/drivers/ComposeDriver.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.drivers
+
+import androidx.compose.ui.semantics.SemanticsProperties
+import androidx.compose.ui.test.*
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidElementStateException
+import io.appium.espressoserver.lib.handlers.exceptions.NoSuchElementException
+import io.appium.espressoserver.lib.helpers.findAllComposeNodeInteractions
+import io.appium.espressoserver.lib.helpers.getNodeInteractionById
+import io.appium.espressoserver.lib.helpers.getSemanticsNode
+import io.appium.espressoserver.lib.model.*
+
+class ComposeDriver : AppDriver {
+    override val name = DriverContext.StrategyType.COMPOSE
+
+    override fun findElement(params: Locator): BaseElement {
+        val parentNodeInteraction =
+            params.elementId?.let { getNodeInteractionById(it) }
+        val nodeInteractions =
+            findAllComposeNodeInteractions(parentNodeInteraction, params)
+
+        if (nodeInteractions.fetchSemanticsNodes(false).isEmpty()) {
+            throw NoSuchElementException(
+                String.format(
+                    "Could not find compose element with strategy '%s' and selector '%s'",
+                    params.using, params.value
+                )
+            )
+        }
+        return ComposeElement(nodeInteractions[0])
+    }
+
+    override fun click(params: AppiumParams): Void? =
+        getNodeInteractionById(params.elementId).performClick().let { return null }
```

Comment:
I'm not quite sure about exceptions here. Let say the given element is stale and does not present in the view anymore. which exception should be thrown by `performClick` then? We must probably intercept it and wrap info StaleElement then.

Reply:
created another method to get non-stale nodes, this can be used in the future for other actions